ui-tests/src/test/java/de/qualicen/maven/ui_tests/ModelEditorTest.java

```diff
@@ -0,0 +1,129 @@
+		
+		driver.get("http://localhost:8080/");
+		
+		if(!login.isLoggedIn()) {
+			login.username("username");
+			login.password("password");
+			login.changeToProject("test-data");
+			login.login(); 
+			assertTrue(login.isLoggedIn());
+		} 
+			
+		// Navigation to requirement
+		projectExplorer.expand("Evaluation");
+		projectExplorer.open("Erlaubnis Autofahren");
+					
+		// Creating and opening new model
+		String modelName = "Model By Automated UI Test " +  new Timestamp(System.currentTimeMillis());
+		requirementOverview.createCEGModelFromRequirement(modelName);
+					
+		// Adding nodes to the CEG
+		WebElement nodeAlter = cegEditor.createNode("Alter", ">17",100,100);//results in x=15, y=60
+		WebElement nodeFS = cegEditor.createNode("Führerschein", "vorhanden",100,300);//results in x=15, y=27
+		WebElement nodeAutofahren = cegEditor.createNode("Autofahren", "erlaubt", 300, 200);
+		
+		// Check if error message is shown (Assert true)
+		assertTrue(cegEditor.errorMessageDisplayed());
+			
+		// Connecting created nodes
+		cegEditor.connectNode(nodeAlter, nodeAutofahren);
+		cegEditor.connectNode(nodeFS, nodeAutofahren);
+		
+		// Check if error message is hidden (Assert false)
+		assertTrue(cegEditor.noWarningsMessageDisplayed());
+		
+		// Last action was creating a connection, so the connection should be removed
+		/*
+		commonControl.undo();
+		
+		// Check if we have only one connection (Assert true)
+		assertTrue(cegEditor.checkUndoConnection());
+		
+		// Redo connecting the two nodes
+		cegEditor.connect(nodeFS, nodeAutofahren); */
+		
+		// Move CEG node
+		//cegEditor.moveCEGNode(nodeFS, 100, 450);
+		
+		// Negate Connection
+		cegEditor.toggleNegateButtonOnLastConnection();
+		
+		// Check if tilde is shown (Assert True)
+		assertTrue(cegEditor.negationDisplayed());
+		
+		// Remove negation 
+		cegEditor.toggleNegateButtonOnLastConnection();
+		
+		// Check if tile is hidden (Assert false)
+		assertFalse(cegEditor.negationDisplayed());
+		
+		// Change connection type
+		cegEditor.changeTypeToOR(nodeAutofahren);
+		cegEditor.changeTypeToAND(nodeAutofahren);
+		
+		assertTrue(cegEditor.correctModelCreated(3, 2));
+					
+		// Save CEG
+		commonControl.save();
+	
+					
+		// Create test specification
+		cegEditor.generateTestSpecification();
+		
+		assertTrue(cegEditor.correctTestSpecificationGenerated(3));
+		
+		// Click on created CEG in the requirement overview
+		cegEditor.clickOnRelatedRequirement("Erlaubnis Autofahren");
+		requirementOverview.clickOnCreatedModel(modelName);
+
+		// Duplicate CEG
+		cegEditor.clickOnRelatedRequirement("Erlaubnis Autofahren");
+		requirementOverview.duplicateCEGModel(modelName);
+		// Click on it, to check if the duplication created a new model
+		requirementOverview.clickOnDuplicateModel(modelName);
+
+		// Delete duplicate
+		cegEditor.clickOnRelatedRequirement("Erlaubnis Autofahren");
+		requirementOverview.deleteDuplicateModel(modelName);
```

Comment:
for this and the next delete: as far as I can see there is no verification that the delete is successfull